File: src/main/java/com/github/stokito/IdeaSingletonInspection/smells/ConstructorsArePrivate.java
package com.github.stokito.IdeaSingletonInspection.smells;

import com.github.stokito.IdeaSingletonInspection.quickFixes.QuickFixes;
import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiMethod;
import com.intellij.psi.PsiModifierList;
import org.jetbrains.annotations.NotNull;

public class ConstructorsArePrivate extends Smell {

  @Override
  public void check(@NotNull final PsiClass aClass) {
    @NotNull final PsiMethod[] constructors = aClass.getConstructors();
    for (final PsiMethod constructor : constructors) {
      @NotNull final PsiModifierList modifiers = constructor.getModifierList();
      if (!modifiers.hasModifierProperty("private")) {
        getHolder().registerProblem(constructor, "Constructor must be private", QuickFixes.CONSTRUCTOR_MODIFIERS);
      }
    }
  }

}

File: src/main/java/com/github/stokito/IdeaSingletonInspection/smells/ConstructorsArePrivate.kt
package com.github.stokito.IdeaSingletonInspection.smells

import com.github.stokito.IdeaSingletonInspection.quickFixes.QuickFixes
import com.intellij.psi.PsiClass
import com.intellij.psi.PsiModifier.PRIVATE

class ConstructorsArePrivate : Smell() {
    override fun check(aClass: PsiClass) {
        val constructors = aClass.constructors
        for (constructor in constructors) {
            val modifiers = constructor.modifierList
            if (!modifiers.hasModifierProperty(PRIVATE)) {
                holder!!.registerProblem(constructor, "Constructor must be private", QuickFixes.CONSTRUCTOR_MODIFIERS)
            }
        }
    }
}